### PR TITLE
update admin_requirement

### PR DESCRIPTION
**Pull request by ramsayleung**
as title

### CODE REVIEW

File: docs/admin_requirement.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+1. all attribute of category could be updated
+
+### 2.5. Getting individual category
+
+### 2.6. Getting all categories
+
+## 3. How to provide
+Category service provides RESTful API to admin-web, here
+is
+[api documentation](https://github.com/reactivesw/category/blob/master/docs/api.md)
```
Comment (kaka89) [inline docs/admin_requirement.md]: use relative path

File: docs/admin_requirement.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Category service is used to manage category. The functions category service
+provides to admin-web as below:
++ create category
++ delete category
++ update category
++ get individual category 
++ get all categories
+
+## 2. Requirement
```
Comment (kaka89) [inline docs/admin_requirement.md]: Requirements

File: docs/admin_requirement.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Admin-Web Requirement on Category
```
Comment (kaka89) [inline docs/admin_requirement.md:1]: delete this

File: docs/admin_requirement.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+This document describes the requirement of admin-web on category service.
+
+## 1. Introduction
```
Comment (kaka89) [inline docs/admin_requirement.md:4]: use "#" not "##"

File: docs/admin_requirement.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+### 2.4 Updating category
+1. all attribute of category could be updated
+
+### 2.5. Getting individual category
+
+### 2.6. Getting all categories
+
+## 3. How to provide
+Category service provides RESTful API to admin-web, here
```
Comment (kaka89) [inline docs/admin_requirement.md]: not to admin-web, maybe "client"? do not specify your service consumer.

File: docs/admin_design.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
++ create a new category
++ delete category
++ update attribute of category
++ get individual category 
++ get all categories in order
+
+## 2. Workflow
+
+### 2.1. Creating a new category
+![create a new category](./resources/create_a_new_category.png)
```
Comment (kaka89) [inline docs/admin_design.md]: Use words to describe your work flow first, other people can not modify your image

File: docs/admin_design.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+2. set parent and ancestor for this new category. 
+
+3. in setp 2, if its parent id is null, it set its parent as empty(thus, it is a
+   root category), otherwise, just setting parent for this new category.Be
+   careful, if the parent of this new category is not a root category, the new
+   category will have grandparent category, its parent and grandparent should be
+   set as ancestor. For example: if you want to create new category named `cpu`,
+   then its parent category is `computer`, and `computer` has a parent category
+   named `electronic product`
+
```
Comment (davis999) [inline docs/admin_design.md]: 1. missing ancestors definition. ancestors is a list of string , store the path from root category to parent category.
2. step 3 is for what？sub step of step 2? 
3. what heppen if change parent? should change it's ancestors? should change sub category's ancestors?

File: docs/admin_design.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+## 3. Workflow
+
+### 3.1. Create Category
+1. get categoryDraft object from admin-web.
+2. get parent id from categoryDraft object and get all categories derived from a
+   same parent.
+3. convert categoryDraft object to category entity object.
+4. call save category function
+5. convert entity object to view object, and return to admin-web.
+
```
Comment (davis999) [inline docs/admin_design.md]: 1. missing set parent id and ancestors workflow.
2. missing check the name on the same level.

File: docs/admin_design.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+3. get all sub categories of this category.
+4. delete this category and its all sub categories
+
+### 3.3 Update Category
+
+1. get category id and update action list from admin-web.
+2. get category entity object by id.
+3. update category by the attribute defined in update action list.
+4. convert entity object to view object, and return to admin-web.
+
```
Comment (davis999) [inline docs/admin_design.md]: check version.

File: docs/admin_design.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+3. convert categoryDraft object to category entity object.
+4. call save category function
+5. convert entity object to view object, and return to admin-web.
+
+### 3.2 Delete Category
+1. get category id and version from admin-web.
+2. get category entity object by id.
+3. get all sub categories of this category.
+4. delete this category and its all sub categories
+
```
Comment (davis999) [inline docs/admin_design.md]: check version.

File: docs/admin_design.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+   category, the new category will have grandparent category, its parent and
+   grandparent should be set as ancestor. For example: if you want to create new
+   category named `cpu`, then its parent category is `computer`, and `computer`
+   has a parent category named `electronic product`
+
+	``` electronic product -> computer -> cpu ```
+	
+	so `electronic product` is a root category, `electronic product` and
+    `computer` are ancestor of `cpu` category
+
```
Comment (davis999) [inline docs/admin_design.md]: use [`electronic product`, `computer`] instead of `electronic product` and `computer`.

**Pull request merged**